x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/snapshot_list/snapshot_details/snapshot_details.tsx

```diff
@@ -178,6 +182,55 @@ const SnapshotDetailsUi: React.FunctionComponent<Props> = ({
             />
           </EuiButtonEmpty>
         </EuiFlexItem>
+
+        {snapshotDetails ? (
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup alignItems="center">
+              <EuiFlexItem grow={false}>
+                <SnapshotDeleteProvider>
+                  {deleteSnapshotPrompt => {
+                    return (
+                      <EuiButtonEmpty
+                        color="danger"
+                        data-test-subj="srSnapshotDetailsDeleteActionButton"
+                        onClick={() =>
+                          deleteSnapshotPrompt(
+                            [{ repository: repositoryName, snapshot: snapshotId }],
+                            onSnapshotDeleted
+                          )
+                        }
+                      >
+                        <FormattedMessage
+                          id="xpack.snapshotRestore.snapshotDetails.deleteButtonLabel"
+                          defaultMessage="Delete"
+                        />
+                      </EuiButtonEmpty>
+                    );
+                  }}
+                </SnapshotDeleteProvider>
+              </EuiFlexItem>
+
+              <EuiFlexItem grow={false}>
+                <EuiButton
+                  href={`#${BASE_PATH}/restore_snapshot/${repositoryName}/${snapshotId}`}
+                  fill
+                  color="primary"
+                  isDisabled={
+                    !(
```

Comment:
Personally I would find this a bit easier to reason about if the negation was within each comparison:

```js
(
  snapshotDetails.state !== SNAPSHOT_STATE.SUCCESS &&
  snapshotDetails.state !== SNAPSHOT_STATE.PARTIAL
)
```